test/org/sagebionetworks/bridge/upload/UploadValidationTaskTest.java

```diff
@@ -150,6 +244,130 @@ public void writeValidationStatusException() {
         verify(task).logWriteValidationStatusException(UploadStatus.SUCCEEDED, toThrow);
     }
 
+    @Test
+    public void dedupeInformation() {
+        // input
+        DynamoStudy study = TestUtils.getValidStudy(UploadValidationTaskTest.class);
+
+        DynamoUpload2 upload2 = new DynamoUpload2();
+        upload2.setUploadId("test-upload");
+
+        UploadValidationContext ctx = new UploadValidationContext();
+        ctx.setStudy(study);
+        ctx.setUpload(upload2);
+
+        // set up validation task
+        UploadValidationTask task = spy(new UploadValidationTask(ctx));
+        task.setHandlerList(handlerList);
+        task.setHealthDataService(healthDataService);
+
+        // execute
+        task.run();
+
+        // verify log helper was called
+        verify(task).logDuplicateUploadRecords(eq(TEST_RECORD_DUPE_LIST_NORMAL));
+    }
+
+    @Test
+    public void dedupeWithoutDuplicate() {
+        // input
+        DynamoStudy study = TestUtils.getValidStudy(UploadValidationTaskTest.class);
+
+        DynamoUpload2 upload2 = new DynamoUpload2();
+        upload2.setUploadId("test-upload");
+
+        UploadValidationContext ctx = new UploadValidationContext();
+        ctx.setStudy(study);
+        ctx.setUpload(upload2);
+
+        // set up validation task
+        UploadValidationTask task = spy(new UploadValidationTask(ctx));
+        task.setHandlerList(handlerList);
+        // only return one test record
+        healthDataService = mock(HealthDataService.class);
+        when(healthDataService.getRecordById(any())).thenReturn(TEST_RECORD);
+        when(healthDataService.getRecordsByHealthcodeCreatedOnSchemaId(any(), any(), any())).thenReturn(Arrays.asList(TEST_RECORD));
+        task.setHealthDataService(healthDataService);
+
+        // execute
+        task.run();
+
+        // verify log helper was NOT called
+        verify(task, times(0)).logDuplicateUploadRecords(any());
+    }
+
+    @Test
+    public void dedupeNullRecord() {
+        // input
+        DynamoStudy study = TestUtils.getValidStudy(UploadValidationTaskTest.class);
+
+        DynamoUpload2 upload2 = new DynamoUpload2();
+        upload2.setUploadId("test-upload");
+
+        UploadValidationContext ctx = new UploadValidationContext();
+        ctx.setStudy(study);
+        ctx.setUpload(upload2);
+
+        // set up validation task
+        UploadValidationTask task = spy(new UploadValidationTask(ctx));
+        task.setHandlerList(handlerList);
+        // return a null record
+        healthDataService = mock(HealthDataService.class);
+        when(healthDataService.getRecordById(any())).thenReturn(null);
+        task.setHealthDataService(healthDataService);
+        task.run();
+        verify(task).logClassNotFoundException(any());
```

Comment:
Would it be more useful to have a verify that logDuplicateUploadRecords *isn't* called?